Client/Frontend/Fakespot/Views/FakespotAdView.swift

```diff
@@ -0,0 +1,315 @@
+        starRatingView.isAccessibilityElement = true
+        let rating = String(format: "%.1f", productAdsData.adjustedRating)
+        starRatingView.accessibilityLabel = String(format: .Shopping.AdjustedRatingStarsAccessibilityLabel, rating)
+
+        let productLinkButtonViewModel = LinkButtonViewModel(
+            title: productAdsData.name,
+            a11yIdentifier: viewModel.productTitleA11yId,
+            fontSize: UX.labelFontSize,
+            contentInsets: UX.linkInsets
+        )
+        productLinkButton.configure(viewModel: productLinkButtonViewModel)
+        gradeReliabilityScoreView.configure(grade: productAdsData.grade)
+        Task {
+            await productImageView.loadImage(from: productAdsData.imageUrl)
+        }
+
+        let cardModel = ShadowCardViewModel(view: contentStackView, a11yId: viewModel.cardA11yId)
+        cardContainer.configure(cardModel)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private var starRatingHeightConstraint: NSLayoutConstraint?
+    private var productImageHeightConstraint: NSLayoutConstraint?
+    private var productImageWidthConstraint: NSLayoutConstraint?
+
+    // MARK: Layout Setup
+    private func setupConstraints() {
+        if !self.subviews.contains(cardContainer) {
+            addSubview(cardContainer)
+            insertSubview(footerLabel, belowSubview: cardContainer)
+        }
+        // normal setup
+        let starsHeight = min(UIFontMetrics.default.scaledValue(for: UX.starSize), UX.starMaxSize)
+        let imageWidth = min(UIFontMetrics.default.scaledValue(for: UX.productImageMinWidth), UX.productImageMaxWidth)
+        let imageHeight = min(UIFontMetrics.default.scaledValue(for: UX.productImageMinHeight), UX.productImageMaxHeight)
+        starRatingHeightConstraint = starRatingView.heightAnchor.constraint(equalToConstant: starsHeight)
+        starRatingHeightConstraint?.isActive = true
+
+        productImageHeightConstraint = productImageView.heightAnchor.constraint(equalToConstant: imageHeight)
+        productImageHeightConstraint?.isActive = true
+
+        productImageWidthConstraint = productImageView.widthAnchor.constraint(equalToConstant: imageWidth)
+        productImageWidthConstraint?.isActive = true
+
+        NSLayoutConstraint.activate([
+            cardContainer.leadingAnchor.constraint(equalTo: leadingAnchor),
+            cardContainer.topAnchor.constraint(equalTo: topAnchor),
+            cardContainer.trailingAnchor.constraint(equalTo: trailingAnchor),
+            cardContainer.bottomAnchor.constraint(equalTo: footerLabel.topAnchor, constant: -UX.vStackSpacing),
+
+            footerLabel.leadingAnchor.constraint(equalTo: leadingAnchor),
+            footerLabel.trailingAnchor.constraint(equalTo: trailingAnchor),
+            footerLabel.bottomAnchor.constraint(equalTo: bottomAnchor),
+        ])
+
+        setupLayout()
```

Comment:
We should call `adjustLayout` here.

Reply:
Naming seems to be inverse. @petruSt24 `setupConstraints` should be `setupLayout` and `setupLayout` shold be named `adjustLayout` to keep the consistency with the other views.